src/stream/src/common/builder.rs

```diff
@@ -122,34 +138,22 @@ impl StreamChunkBuilder {
             self.column_builders[i + self.matched_start_pos].append_datum(&None)?;
         }
 
-        Ok(())
+        self.inc_size()
     }
 
     /// append a row with matched value and fill the coming side with null.
-    pub fn append_row_matched(&mut self, op: Op, row_matched: &Row) -> Result<()> {
+    ///
+    /// A [`StreamChunk`] will be returned when `size == capacity`
+    pub fn append_row_matched(&mut self, op: Op, row_matched: &Row) -> Result<Option<StreamChunk>> {
         self.ops.push(op);
         for i in 0..self.column_builders.len() - row_matched.size() {
             self.column_builders[i + self.update_start_pos].append_datum_ref(None)?;
         }
         for i in 0..row_matched.size() {
             self.column_builders[i + self.matched_start_pos].append_datum(&row_matched[i])?;
         }
-        Ok(())
-    }
-
-    pub fn finish(self) -> Result<StreamChunk> {
-        let new_arrays = self
-            .column_builders
-            .into_iter()
-            .map(|builder| builder.finish())
-            .collect::<Result<Vec<_>>>()?;
-
-        let new_columns = new_arrays
-            .into_iter()
-            .map(|array_impl| Column::new(Arc::new(array_impl)))
-            .collect::<Vec<_>>();
 
-        Ok(StreamChunk::new(self.ops, new_columns, None))
+        self.inc_size()
     }
 
     pub fn take(&mut self) -> Result<Option<StreamChunk>> {
```

Comment:
Should this method be `pub` now?

Reply:
`pub take()` is called by HashJoinExecutor to generate a chunk from the remainder in the chunk builder.
https://github.com/singularity-data/risingwave/blob/35093ccddac4d688092a6162b780bdbac4a6adb7/src/stream/src/executor/hash_join.rs#L573